Named inference rule for `abs`.

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #22308 More named inference rules for pointwise unary ops
* #22268 Named inference rule for more pointwise ops.
* #22267 Enable named inference for some unary pointwise ops
* #22223 Implement unify_from_right
* **#22151 Named inference rule for `abs`.**

Adds a named inference rule for `abs`. This is an example of how to add
a named inference rule for a function that simply propagates names on
its inputs to the outputs.
- Actually, this diff adds named inference rules for other functions as
well, but they are not enabled yet (due to named_guard=False for those
functions). Those will be enabled in a future change.

Test Plan:
- `python test/test_namedtensor.py` [namedtensor ci]

gh-metadata: pytorch pytorch 22151 gh/zou3519/61/head

Differential Revision: [D15970326](https://our.internmc.facebook.com/intern/diff/D15970326)

zou3519 committed:
Named inference rule for `abs`.

Adds a named inference rule for `abs`. This is an example of how to add
a named inference rule for a function that simply propagates names on
its inputs to the outputs.
- Actually, this diff adds named inference rules for other functions as
well, but they are not enabled yet (due to named_guard=False for those
functions). Those will be enabled in a future change.

Test Plan:
- `python test/test_namedtensor.py` [namedtensor ci]

zou3519 committed:
Update on "Named inference rule for `abs`."

Named inference rule for `abs`.

Adds a named inference rule for `abs`. This is an example of how to add
a named inference rule for a function that simply propagates names on
its inputs to the outputs.
- Actually, this diff adds named inference rules for other functions as
well, but they are not enabled yet (due to named_guard=False for those
functions). Those will be enabled in a future change.

Test Plan:
- `python test/test_namedtensor.py` [namedtensor ci]

gh-metadata: pytorch pytorch 22151 gh/zou3519/61/head

zou3519 committed:
Update on "Named inference rule for `abs`."

Named inference rule for `abs`.

Adds a named inference rule for `abs`. This is an example of how to add
a named inference rule for a function that simply propagates names on
its inputs to the outputs.
- Actually, this diff adds named inference rules for other functions as
well, but they are not enabled yet (due to named_guard=False for those
functions). Those will be enabled in a future change.

Test Plan:
- `python test/test_namedtensor.py` [namedtensor ci]

gh-metadata: pytorch pytorch 22151 gh/zou3519/61/head

nairbv commented on aten/src/ATen/native/UnaryOps.cpp:
```diff
@@ -137,6 +140,15 @@ Tensor& _sigmoid_out_cpu(Tensor& result, const Tensor& self) {
+    if (self.is_named()) {                                      \
+      at::internal_set_names_inplace(result, self.names());     \
+    }
+#else
```
Would it be better to have a perform_name_inference(result, self) function that checks the namedtensor_enabled guard in the implementation (and is otherwise a no-op)? Seems like then we'd have fewer macros, could be cleaner once we eventually get rid of the namedtensor_enabled guard, we could use in cases where the result/self names differ, and would be clearer what's going on / whats' being passed as arguments in context.

zou3519 replied:
Sure, that would look nicer, thanks for the suggestion

zou3519 replied:
Changed.

zou3519 committed:
Update on "Named inference rule for `abs`."

Named inference rule for `abs`.

Adds a named inference rule for `abs`. This is an example of how to add
a named inference rule for a function that simply propagates names on
its inputs to the outputs.
- Actually, this diff adds named inference rules for other functions as
well, but they are not enabled yet (due to named_guard=False for those
functions). Those will be enabled in a future change.

Test Plan:
- `python test/test_namedtensor.py` [namedtensor ci]

gh-metadata: pytorch pytorch 22151 gh/zou3519/61/head

zou3519 committed:
Update on "Named inference rule for `abs`."

Named inference rule for `abs`.

Adds a named inference rule for `abs`. This is an example of how to add
a named inference rule for a function that simply propagates names on
its inputs to the outputs.
- Actually, this diff adds named inference rules for other functions as
well, but they are not enabled yet (due to named_guard=False for those
functions). Those will be enabled in a future change.

Test Plan:
- `python test/test_namedtensor.py` [namedtensor ci]

gh-metadata: pytorch pytorch 22151 gh/zou3519/61/head

nairbv reviewed (approved):
looks clearer to me thanks

zou3519 committed:
Update on "Named inference rule for `abs`."

Named inference rule for `abs`.

Adds a named inference rule for `abs`. This is an example of how to add
a named inference rule for a function that simply propagates names on
its inputs to the outputs.
- Actually, this diff adds named inference rules for other functions as
well, but they are not enabled yet (due to named_guard=False for those
functions). Those will be enabled in a future change.

Test Plan:
- `python test/test_namedtensor.py` [namedtensor ci]

gh-metadata: pytorch pytorch 22151 gh/zou3519/61/head
